### DISCUSSION TITLE
How can I set different authentication header for different swagger calls

### DISCUSSION THREAD

**Discussion by BernhardJ** [Burgyn/MMLib.SwaggerForOcelot#241] [Q&A]
Hi

First, yes i have read the issue #83 concerning authentication headers.
In my config, I have two or more SwaggerEndPoints.
Each SwaggerEndPoint is secured by BasicAuthentication. Each Endpoint uses anothe username / password.

During configuration (UseSwaggerForOcelotUI) I can add an array of KeyValuePair.
If I add a key Authorization an the value Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==, this is used for all swagger endpoints.

Is there a possibility to configure the key / value pair (array) for each swagger endpoint?
Or is there a possibility to add the header just before sending the request to the downstream swagger endpoint?

Thanks for answering my questions.

**Comment by Burgyn** [comment]
Hi @BernhardJ,

Unfortunately, there is no direct way to do this. But it occurs to me that you can implement your own `IDownstreamSwaggerDocsRepository` and register this to the DI Container.
```csharp
services.AddTransient<IDownstreamSwaggerDocsRepository, YourOwnDownstreamSwaggerDocsRepository>();
```

You can find the [original implementation here.](https://github.com/Burgyn/MMLib.SwaggerForOcelot/blob/master/src/MMLib.SwaggerForOcelot/Repositories/DownstreamSwaggerDocsRepository.cs)

However, you can still create a pull request and improve this package 😉
